proposals/lip-0051.md

```diff
@@ -2551,7 +2551,7 @@ def verifyCrossChainMessage(trs: Transaction, ccm: CCM) -> None:
 def beforeCrossChainCommandExecution(trs: Transaction, ccm: CCM) -> None:
     relayerAddress = sha256(trs.senderPublicKey)[:ADDRESS_LENGTH]
 
-    messageFeeTokenID = Interoperability.getMessageFeeTokenID(ccm.receivingChainID)
```

Comment:
Question to make sure I understand: That means that our previous protocol was wrong because we are already in the receiving chain while running this code, i.e., `ccm.receivingChainID = OWN_CHAIN_ID`, right?

Btw, while checking this I had a look in the `getMessageFeeTokenID` function and it seems to me that it does not have the right functionality for sidechains in case it gets as input `OWN_CHAIN_ID`. See my comment in LIP 45.

Reply:
yeah. Im wondering if using the sending chain is 'secure' in this sense, or if we should try to get it from the receiving chain in case that fails

Reply:
Yep. In the "normal case" of direct channel communication, we are always running this in the receiving chain, so we should be checking the sending chain channel. 

For the forwarding it is not that important I guess, this is why we did not pay much attention here: Since all channels between mainchain and sidechains have the same message fee token (no matter if is LSK or anything else in the future), then checking sending/receiving chain should give the same result, so sending chain is also fine. Perhaps we should document that it is very important to maintain this property in the future (i.e., "all channels with mainchain should use the same message fee token"), this is something that auditors were asking I guess..